File: src/main/java/org/retrolauncher/backend/app/platforms/infrastructure/desktop/controllers/PlatformController.java
package org.retrolauncher.backend.app.platforms.infrastructure.desktop.controllers;

import org.retrolauncher.backend.app.platforms.application.exceptions.CoreFolderNotFoundException;
import org.retrolauncher.backend.app.platforms.application.usecases.UpdatePlatformsListUseCase;

public class PlatformController {
    private final UpdatePlatformsListUseCase updatePlatformsListUseCase;

    public PlatformController(UpdatePlatformsListUseCase updatePlatformsListUseCase) {
        this.updatePlatformsListUseCase = updatePlatformsListUseCase;
    }

    public void updateList() throws CoreFolderNotFoundException {
        this.updatePlatformsListUseCase.execute();
    }
}

File: backend/src/app/platform/delivery/desktop/platform_controller/platform_controller.go
package platform_controller

import (
	"retrolauncher/backend/src/app/platform/application/create_platform"
	"retrolauncher/backend/src/app/platform/application/get_platform_types"
	list_platforms "retrolauncher/backend/src/app/platform/application/list-platforms"
)

type PlatformController struct {
	createPlatform   *create_platform.CreatePlatform
	getPlatformTypes *get_platform_types.GetPlatformTypes
	listPlatforms    *list_platforms.ListPlatforms
}

func New(
	createPlatform *create_platform.CreatePlatform,
	getPlatformTypes *get_platform_types.GetPlatformTypes,
	listPlatforms *list_platforms.ListPlatforms,
) *PlatformController {
	return &PlatformController{
		createPlatform:   createPlatform,
		getPlatformTypes: getPlatformTypes,
		listPlatforms:    listPlatforms,
	}
}

func (c *PlatformController) Create(input CreateInputDto) []error {
	return c.createPlatform.Execute(create_platform.Input{
		Name:         input.Name,
		PlatformType: input.PlatformType,
		Path:         input.Path,
	})
}

func (c *PlatformController) GetPlatformTypes() []string {
	return *c.getPlatformTypes.Execute()
}

func (c *PlatformController) List() ([]*list_platforms.Output, error) {
	return c.listPlatforms.Execute()
}

type CreateInputDto struct {
	Name         string `json:"name"`
	PlatformType string `json:"type"`
	Path         string `json:"path"`
}
